packages/shared/src/components/modals/SharePostModal.tsx

```diff
@@ -26,18 +27,26 @@ export default function PostModal({
   ...props
 }: PostModalProps): ReactElement {
   const { showArticleOnboarding } = useContext(OnboardingContext);
-  const { post, isLoading } = usePostById({ id, isFetchingNextPage });
+  const {
+    post,
+    query: { isLoading, isFetching },
+  } = usePostById({ id });
+  const isPostLoading = isLoading || isFetchingNextPage || isFetching;
```

Comment:
Just a thought is there no way we can directly export this from the `usePostbyId` fn as we reuse it two times like this? 
Just to avoid changes in one place down the line.

Reply:
I agree with your thoughts. I tried that first, I extracted some of these properties and exposed it down from the hook, then another one. Thought of using `Pick<UseQueryResult, 'property'>` for the returned type but as it grows it just becomes too much to do it one by one every time, IMHO.
It made me realize to just return the query itself so if we will have another thing to destructure from the query, we won't have to update the hook by what is already available from the query itself.

But that is definitely an option we can refactor to. WDYT?

Reply:
Hmm, well the concern is that it makes it very opinionated.
You know today that you need all 3 to evaluate, but what happens if Ante needs to work on this next week, he  might have no clue about that constraint or why it's needed.

Thus I think the more verbose the hook is, the better it will be.
CC: @capJavert Would love to get your thoughts on this as well.

Reply:
No, that's a good point. There are pros and cons in either of the implementations, this would probably boil down to what we prefer more in the long run.

I am fine with any of the two.

Reply:
I agree with points Chris made to make it more verbose.

On another note we might be able to just use `isLoading` here. From what I see we need to make sure post data is available, `isLoading` will only be `false` until initial fetch is done, after that it is going to be `true`. So I am curios if we even need to check `isFetching` and `isFetchingNextPage` in addition to `isLoading` in all of the usages?

Reply:
>  So I am curios if we even need to check isFetching and isFetchingNextPage in addition to isLoading in all of the usages?

The `isLoading` is always `false` from the beginning until the initial fetch is done, and we need to show a loading state on the initial load. That's the reason why we need the `isFetching`. We will need `isFetchingNextPage` when the user is switching left and right, though we can probably omit this just haven't tried yet.

Logging `isLoading`:
![image](https://user-images.githubusercontent.com/13744167/231417013-f29167df-b815-4d08-a9e2-0ad09f395ccc.png)

Alright. Thank you both for your thoughts. Let's go with the verbose implementation.

Reply:
Thanks for explaining cases for why fetching is needed. I would advise to then call `isPostLoading` something different and also verbose so we know it does two things. This is more of a problem of react-query nomenclature I think.

Reply:
`isPostLoadingOrFetching` is fine with me, but not everyone like that kind of verbosity.

Reply:
> isPostLoadingOrFetching is fine with me, but not everyone like that kind of verbosity.

We can do this. Better to be clear than miss it. Thanks!